mlflow/cli.py

```diff
@@ -84,8 +84,14 @@ def cli():
               help="If specified, the given run ID will be used instead of creating a new run. "
                    "Note: this argument is used internally by the MLflow project APIs "
                    "and should not be specified.")
+@click.option("--docker-auth-config", metavar="MLFLOW_DOCKER_AUTH_CONFIG",
```

Comment:
Is this necessary? Doesn't the docker python client use your local docker daemon's authentication information already?

Reply:
I think it's necessary because sometimes we may need to push to different repositories. As you said above, users may have different needs, like different registries per team. So thinking about these use cases, we need to pass credentials.
What do you think?

Reply:
Complementing, as I said above, regarding the `job_namespace`, I think the same applies to publish images. So the registry and image namespace would be in MLproject file.

Reply:
Would you need this functionality yourself? I feel the common case is that the docker daemon's authentication information is enough for most use cases. As such, I think it's better to widen the API when more concrete use cases come up in the future.